pkg/controller/ambassadorinstallation/reconcile.go

```diff
@@ -308,14 +318,24 @@ func (r *ReconcileAmbassadorInstallation) Reconcile(request reconcile.Request) (
 					Version: "v2",
 					Kind:    "RateLimitService",
 				}, request.Namespace)
+
 				if err != nil {
 					log.Error(err, "Could not look up RateLimitService in the cluster")
-					return reconcile.Result{}, err
```

Comment:
Why are we not returning when there is an error?

Reply:
Ok, replaced by a new `Condition` and error.